### PR TITLE
KTOR-5181 `RoutingContext` should not be `CoroutineScope`

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

**Review by e5l — CHANGES_REQUESTED** [review]
Could you please check the CI?

**Comment by e5l** [comment]
Hey @bjhham, could you please take a look? I will take this to fix while @rsinukov is on vacation

File: ktor-client/ktor-client-plugins/ktor-client-call-id/jvmAndNix/test/io/ktor/client/plugins/callid/CallIdTest.kt [hunk 1]
```diff
@@ -12,6 +12,7 @@ import io.ktor.server.plugins.callid.*
 import io.ktor.server.response.*
 import io.ktor.server.routing.*
 import io.ktor.server.testing.*
+import kotlin.coroutines.*
```
Comment (bjhham) [inline ktor-client/ktor-client-plugins/ktor-client-call-id/jvmAndNix/test/io/ktor/client/plugins/callid/CallIdTest.kt:15]: Looks like an unused import here.

File: ktor-server/ktor-server-tests/jvmAndNix/test/io/ktor/tests/server/plugins/CallIdTest.kt [hunk 2]
```diff
@@ -15,6 +15,7 @@ import io.ktor.server.response.*
 import io.ktor.server.routing.*
 import io.ktor.server.testing.*
 import io.ktor.util.pipeline.*
+import kotlin.coroutines.*
```
Comment (bjhham) [inline ktor-server/ktor-server-tests/jvmAndNix/test/io/ktor/tests/server/plugins/CallIdTest.kt:18]: Possibly here too.

File: ktor-client/ktor-client-plugins/ktor-client-call-id/jvmAndNix/test/io/ktor/client/plugins/callid/CallIdTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-plugins/ktor-client-call-id/jvmAndNix/test/io/ktor/client/plugins/callid/CallIdTest.kt:15]: it use for accessing `coroutineContext` of the suspend function

Comment (bjhham) [reply ktor-client/ktor-client-plugins/ktor-client-call-id/jvmAndNix/test/io/ktor/client/plugins/callid/CallIdTest.kt:15]: Ah I see it was a compile error in CI before 💡

**Review by bjhham — APPROVED** [review]

**Comment by bjhham** [comment]
Looks like there's a problem with how my retry test magic works, I'll look into it 😞 

And it seems like the allocation tests fail sometimes, might need more warmup or something... 🤔

**Review by e5l — APPROVED** [review]

**Pull request merged**